android/java/org/chromium/chrome/browser/crypto_wallet/fragments/onboarding_fragments/BackupWalletFragment.java

```diff
@@ -5,42 +5,229 @@
+
+                return;
+            }
+            KeyringService keyringService = getKeyringService();
+            if (keyringService != null) {
+                final String passwordToUse = mPasswordFromBiometric.isEmpty()
+                        ? mBackupWalletPassword.getText().toString()
+                        : mPasswordFromBiometric;
+                keyringService.getMnemonicForDefaultKeyring(passwordToUse, result -> {
+                    if (!result.isEmpty()) {
+                        mOnboardingViewModel.setPassword(passwordToUse);
+                        onNextPage.gotoNextPage(false);
+                    } else {
+                        showPasswordRelatedControls(true);
+                        mBackupWalletPassword.setError(
+                                getString(R.string.incorrect_password_error));
+                    }
+                });
             }
         });
+        mBackupWalletCheckbox.setOnCheckedChangeListener((buttonView, isChecked) -> {
+            if (!mOnboardingLayout && !mBiometricExecuted
+                    && Build.VERSION.SDK_INT >= Build.VERSION_CODES.P
+                    && Utils.isBiometricAvailable(getContext())
+                    && KeystoreHelper.shouldUseBiometricOnUnlock()) {
+                createBiometricPrompt();
+
+                return;
+            }
+            enableDisableContinueButton(isChecked);
+        });
         TextView backupWalletSkipButton = view.findViewById(R.id.btn_backup_wallet_skip);
         backupWalletSkipButton.setOnClickListener(v -> onNextPage.gotoNextPage(true));
+        mBiometricBackupWalletImage.setOnClickListener(v -> {
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P
+                    && Utils.isBiometricAvailable(getContext())) {
+                showPasswordRelatedControls(false);
+                createBiometricPrompt();
+            }
+        });
+        checkOnBiometric();
+    }
+
+    private void checkOnBiometric() {
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.P
+                || !KeystoreHelper.shouldUseBiometricOnUnlock()
+                || !Utils.isBiometricAvailable(getContext())) {
+            showPasswordRelatedControls(true);
+
+            return;
+        }
+    }
+
+    private void showPasswordRelatedControls(boolean show) {
+        if (mOnboardingLayout) return;
+
+        int visibility = show ? View.VISIBLE : View.GONE;
+        mBackupWalletTitle.setVisibility(visibility);
+        mBackupWalletPassword.setVisibility(visibility);
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P
+                && Utils.isBiometricAvailable(getContext())
+                && KeystoreHelper.shouldUseBiometricOnUnlock()) {
+            mBiometricBackupWalletImage.setVisibility(visibility);
+        }
+    }
+
+    private void enableDisableContinueButton(boolean isChecked) {
+        if (isChecked
+                && (mOnboardingLayout
+                        || !mOnboardingLayout
```

Comment:
`mOnboardingLayout || !mOnboardingLayout` will always return `true`, either wanna use only one or it is not required?